src/App.tsx

```diff
@@ -1,56 +1,76 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropdownContent }
+  from './components/Dropdown-content/Dropdown-content';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(
+    debounce((value) => {
+      setAppliedQuery(value);
+      setIsDropdownVisible(true);
+    }, 1000),
```

Comment:
Per task - debounce time should be the component property
You will need some refactoring to do it (see comment above)

Reply:
Use callback is unnecessary here